tests/pytests/integration/cli/test_salt_call.py

```diff
@@ -252,16 +256,37 @@ def test_pillar_items_masterless(
         assert ret.json["monty"] == "python"
 
 
-def test_masterless_highstate(salt_call_cli):
+def test_masterless_highstate(salt_call_cli, base_env_state_tree_root_dir, tmp_path):
     """
     test state.highstate in masterless mode
     """
-    destpath = os.path.join(RUNTIME_VARS.TMP, "testfile")
-    ret = salt_call_cli.run("--local", "state.highstate")
-    assert ret.exitcode == 0
-    state_run_dict = next(iter(ret.json.values()))
-    assert state_run_dict["result"] is True
-    assert state_run_dict["__id__"] == destpath
+    top_sls = """
+    base:
+      '*':
+        - core
+        """
+
+    testfile = tmp_path / "testfile"
+    core_state = """
+    {}:
+      file:
+        - managed
+        - source: salt://testfile
+        - makedirs: true
+        """.format(
+        testfile
+    )
+
+    expected_id = "{}".format(testfile)
```

Comment:
```suggestion
    expected_id = str(testfile)
```
?
Feel free to say, no 😁